var/spack/repos/builtin/packages/atlas/package.py

```diff
@@ -92,10 +95,8 @@ def install(self, spec, prefix):
 
         # Lapack resource to provide full lapack build. Note that
         # ATLAS only provides a few LAPACK routines natively.
-        lapack_stage = self.stage[1]
-        lapack_tarfile = os.path.basename(lapack_stage.fetcher.url)
-        lapack_tarfile_path = join_path(lapack_stage.path, lapack_tarfile)
-        options.append('--with-netlib-lapack-tarfile=%s' % lapack_tarfile_path)
+        options.append('--with-netlib-lapack-tarfile=%s' %
+                       self.stage[1].archive_file)
```

Comment:
> Would it be possible to modify the composite class so that we can refer to resource stages by name?

Yes, it would and it makes a lot of sense. In theory the purpose of the composite is not to expose the components. What is done here is an "hack" because the package wants to know where the `tgz` of the resource resides and expand it on his own at configure time. I can work on it in another PR. What is the priority you give to this? Do you want it for `v0.10`?